lib/SILGen/SILGenApply.cpp

```diff
@@ -1786,6 +1786,139 @@ static bool hasUnownedInnerPointerResult(CanSILFunctionType fnType) {
+  SourceLoc loc = magicLiteral->getStartLoc();
+
+  switch (magicLiteral->getKind()) {
+  case MagicIdentifierLiteralExpr::FileIDSpelledAsFile:
+  case MagicIdentifierLiteralExpr::FileID: {
+    std::string value = loc.isValid() ? SGF.getMagicFileIDString(loc) : "";
+    return emitStringLiteral(SGF, magicLiteral, value, C,
+                             magicLiteral->getStringEncoding());
+  }
+
+  case MagicIdentifierLiteralExpr::FilePathSpelledAsFile:
+  case MagicIdentifierLiteralExpr::FilePath: {
+    StringRef value = loc.isValid() ? SGF.getMagicFilePathString(loc) : "";
+    return emitStringLiteral(SGF, magicLiteral, value, C,
+                             magicLiteral->getStringEncoding());
+  }
+
+  case MagicIdentifierLiteralExpr::Function: {
+    StringRef value = loc.isValid() ? SGF.getMagicFunctionString() : "";
+    return emitStringLiteral(SGF, magicLiteral, value, C,
+                             magicLiteral->getStringEncoding());
+  }
+
+  case MagicIdentifierLiteralExpr::Line:
+  case MagicIdentifierLiteralExpr::Column: {
+    SourceLoc Loc = magicLiteral->getStartLoc();
+    unsigned Value = 0;
+    if (Loc.isValid()) {
+      Value = magicLiteral->getKind() == MagicIdentifierLiteralExpr::Line
+                  ? ctx.SourceMgr.getPresumedLineAndColumnForLoc(Loc).first
+                  : ctx.SourceMgr.getPresumedLineAndColumnForLoc(Loc).second;
+    }
+
+    auto silTy = SILType::getBuiltinIntegerLiteralType(ctx);
+    auto ty = silTy.getASTType();
+    SILValue integer = SGF.B.createIntegerLiteral(magicLiteral, silTy, Value);
+    ManagedValue integerManaged = ManagedValue::forUnmanaged(integer);
+    PreparedArguments builtinLiteralArgs;
+    builtinLiteralArgs.emplace(AnyFunctionType::Param(ty));
+    builtinLiteralArgs.add(magicLiteral, RValue(SGF, {integerManaged}, ty));
+    return builtinLiteralArgs;
+  }
+  case MagicIdentifierLiteralExpr::DSOHandle:
+    llvm_unreachable("handled elsewhere");
+  }
+}
+
+static inline PreparedArguments buildBuiltinLiteralArgs(SILGenFunction &SGF,
+                                                        SGFContext C,
+                                                        LiteralExpr *literal) {
+  if (auto stringLiteral = dyn_cast<StringLiteralExpr>(literal)) {
+    return buildBuiltinLiteralArgs(SGF, C, stringLiteral);
+  } else if (auto nilLiteral = dyn_cast<NilLiteralExpr>(literal)) {
+    return buildBuiltinLiteralArgs(nilLiteral);
+  } else if (auto booleanLiteral = dyn_cast<BooleanLiteralExpr>(literal)) {
+    return buildBuiltinLiteralArgs(SGF, C, booleanLiteral);
+  } else if (auto integerLiteral = dyn_cast<IntegerLiteralExpr>(literal)) {
+    return buildBuiltinLiteralArgs(SGF, C, integerLiteral);
+  } else if (auto floatLiteral = dyn_cast<FloatLiteralExpr>(literal)) {
+    return buildBuiltinLiteralArgs(SGF, C, floatLiteral);
+  } else {
+    return buildBuiltinLiteralArgs(
+        SGF, C, dyn_cast<MagicIdentifierLiteralExpr>(literal));
```

Comment:
I think you should either use `cast` or move it into an if check like above and add a `llvm_unreachable` at the end. This is so we crash here if someone adds a new literal expression but doesn’t update this function.